### ISSUE TITLE
Multiple Spaces in a Row Are Ignored when They Equal a Code Block Indicator

### ISSUE THREAD

**Issue by Cito** [platers/obsidian-linter#690]
## Describe the Bug

Sometimes, whitespace is not compacted.

## How to Reproduce

I am using Linter 1.12.0 with Obsidian 1.2.3 on Windows.

And the following settings:
- Remove multiple spaces: on

You can also set these, but they don't seem to matter here:
- Space after list markers: on
- Unorderderd list style: on
- Convert bullet list markers: on

Copy the following Markdown code:
```
- foo
-     bar    baz
-    bar    baz
```

Note that on the second line, there are 5 blanks before "bar" and 4 blanks before "baz". On the third line there are 4 blanks before both words.

Run "Lint the current file"

Only the two occurences of multiple whitespace in the third line will be compacted, none of those in the second line:
```
- foo
-     bar    baz
- bar baz
```

## Expected Behavior

On the second lines, the two occurences of multiple whitespace should be compacted as well.

Expected output:
```
- foo
- bar baz
- bar baz
```

**Comment by pjkaufman** [comment]
Thanks for reporting this. I can reproduce with the example provided. The regex matches just fine, but I need to check why it is not actually updating the data as expected. It could be that one of the ignores is catching the second foo bar. It could be getting considered to be a code block or something like that. If that is the case, I am not sure if I can do anything about it.

**Comment by pjkaufman** [comment]
So the problem is that 4 spaces like you have in the example is an indicator of a code block. Thus the space cannot be removed as the rule ignores code blocks to prevent issues with spacing in code blocks.

Here is what it looks like before the regex runs due to the ignores:
``` markdown
 - foo
 - {CODE_BLOCK_PLACEHOLDER}
 -    bar    baz
```

As far as I can tell, this is expected. I will just double check with the online AST parser and see if that still comes out to be a code block reference. If it is, then this is expected behavior.

**Comment by pjkaufman** [comment]
The AST does interpret that block as a code block:
![image](https://user-images.githubusercontent.com/29898683/232782240-d3eecb31-9f08-406d-8b45-429ac4ee6967.png)

So I am not sure this can be fixed if we preserve spacing in code blocks.

I can add an example or more documentation around this if you would like @Cito . But other than that, I plan to close this as a non-issue as it is expected behavior for the amount of spaces present.

**Comment by pjkaufman** [comment]
Feel free to respond on the issue, but I am closing it as it worls as intended.
